src/components/PeopleFilters.tsx

```diff
@@ -1,17 +1,117 @@
+import classNames from 'classnames';
+import {
+  useQueryParams,
+  StringParam,
+  withDefault,
+  ArrayParam,
+} from 'use-query-params';
+
+import { SexEnum } from '../types/SexEnum';
+
+const MyFiltersParam = withDefault(ArrayParam, []);
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useQueryParams({
+    sex: StringParam,
+    century: MyFiltersParam,
+    query: StringParam,
+  });
+
+  const { century, sex, query } = searchParams;
+
+  const handleCenturyFilter = (
+    cent: string | null,
+  ) => (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    e.preventDefault();
+
+    if (!cent) {
+      setSearchParams({
+        century: undefined,
+      });
+
+      return;
+    }
+
+    if (!century.includes(cent as never)) {
+      setSearchParams({
+        century: [...century, cent],
+      });
+
+      return;
+    }
+
+    setSearchParams({
+      century: century.filter(c => c !== cent),
+    });
+  };
+
+  const handleSexChange = (
+    newSex: string | null,
+  ) => (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    e.preventDefault();
+
+    if (!newSex) {
+      setSearchParams({
+        sex: undefined,
+      });
+
+      return;
+    }
+
+    setSearchParams({
+      sex: newSex,
+    });
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value
+      .toLowerCase()
+      .replace(/[^a-zA-Z\s\u0400-\u04FF]/gu, '');
+
+    setSearchParams({
+      query: value || null,
+    });
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <a
+          className={classNames(
+            { 'is-active': !sex },
+          )}
+          href="#/people"
+          onClick={handleSexChange(null)}
+        >
+          All
+        </a>
```

Comment:
Why didn't you use the `SearchLink` component for these purposes? Refactor it to simplify the logic of this component. You don't really need all those handlers 
![image](https://github.com/mate-academy/react_people-table-advanced/assets/99403198/cc88f978-a28d-4292-86a2-2add92331782)